submissions/iakow/OOP-exercise/index.js

```diff
@@ -0,0 +1,87 @@
+class Inhabitant {
+  constructor({ species, name, gender, legs, hands, say }) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.hands = hands;
+    this.say = say;
+    this.friends = [];
+  }
+
+  makeFriends(...args) {
+    const { friends } = this;
+
+    args.forEach((item) => {
+      if (item instanceof Inhabitant && item != this && !friends.includes(item.name)) {
+        friends.push(item.name);
+      }
+    });
+  }
```

Comment:
Not a KO to fix. Fixing is at your discretion.

`args` is too generic name for the arguments expected. `friends` already in use, but we might distinguish new friends passed as arguments by more specific naming. Sematically speaking naming is important.

Test against `instanceof Inhabitant` can be implemented using `Array.filter`.
Same is true with regards to equality with `this`.

A one-liner to ensure only unique values/objects are in array is `list = [...new Set(list)]`. Check [docs on `Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set), you will find it helpful.